contracts/Hangman.sol

```diff
@@ -15,14 +15,20 @@ contract Hangman is Ownable {
     event TurnTaken();
 
     /*
-     * @notice Initializes the Hangman Game
+     * @notice Initializes the Hangman Game with a solution
+     * @dev only the owner can set the game. This will be done by the factory
      * @param bytes the solution
      * @param uint the maximum allowed number of misses before it's game over
      */
-    constructor(bytes memory _solution, uint _maxAllowedMisses) public {
+    function setSolution(bytes _solution, uint _maxAllowedMisses) onlyOwner external {
+        require(solution.length == 0, "solution already set");
         solution = _solution;
         maxAllowedMisses = _maxAllowedMisses;
-        currentMisses = 0;
+    }
+
+    modifier gameReady() {
+        require(solution.length > 0, "solution not set");
+        _;
```

Comment:
what is this line for?

Reply:
the line is what allows it to be a modifier.